native/src/main/java/io/ballerina/lib/confluent/registry/CustomSchemaRegistryClient.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.confluent.registry;
+
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import org.apache.avro.Schema;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.ballerina.lib.confluent.registry.ModuleUtils.NATIVE_CLIENT;
+import static io.ballerina.lib.confluent.registry.Utils.CLIENT_INVOCATION_ERROR;
+
+public class CustomSchemaRegistryClient {
+
+    public static void generateSchemaRegistryClient(BObject registryClient, BMap<BString, Object> config) {
+        BString baseUrl = (BString) config.get(ModuleUtils.BASE_URL);
+        long identityMapCapacity = config.getIntValue(ModuleUtils.IDENTITY_MAP_CAPACITY);
+        BMap<BString, Object> originals = (BMap<BString, Object>) config.getMapValue(ModuleUtils.ORIGINALS);
+        BMap<BString, Object> httpHeaders = (BMap<BString, Object>) config.getMapValue(ModuleUtils.HEADERS);
+        Map<String, String> configurations = new HashMap<>();
+        for (BString key: originals.getKeys()) {
+            configurations.put(key.getValue().replaceAll("^\"|\"$", ""),
+                               originals.get(key).toString());
+        }
+        Map<String, String> headers = new HashMap<>();
+        for (BString header: httpHeaders.getKeys()) {
+            headers.put(header.getValue(), httpHeaders.get(header).toString());
+        }
+        SchemaRegistryClient registry = new CachedSchemaRegistryClient(baseUrl.getValue(), (int) identityMapCapacity,
+                                                                       configurations, headers);
+        registryClient.addNativeData(NATIVE_CLIENT, registry);
+    }
+
+    public static Object register(BObject registryClient, BString subject, BString schema) {
+        SchemaRegistryClient schemaRegistryClient = (SchemaRegistryClient) registryClient.getNativeData(NATIVE_CLIENT);
+        Schema.Parser parser = new Schema.Parser();
```

Comment:
Do we need to create `Schema.Parser` object every-time we invoke this method ? If this is a reusable object we can create a `static final` variable at the class level and use it. But, its important to check whether this `Schema.Parser` concurrent safe if we do this.

Reply:
In `Schema.Parser` there are some instance variables to check the validation of the schema and these variables can be altered during the invocation of `Schema.Parser` APIs. Therefore there can be incorrect results occurred while executing different methods of a `Schema.Parser` instance concurrently. Therefore it will be better to have a separate instance for each API.

Reply:
If `Schema.Parser` is not thread-safe, then this is the correct approach. Lets proceed with this.